TSV export, only expose .json files, create our own index for directory

ldemailly opened the pull request:
This should let fortio servers be programmatically backed up to cloud storage

ldemailly committed:
TSV export, only expose .json files, create our own index for directory

ldemailly committed:
Make linters happy

ldemailly committed:
Merge branch 'master' into tsvindex

ldemailly committed:
Adding a cache for tsv (kinda overkill but one less todo)

ldemailly committed:
Merge branch 'master' into tsvindex

ldemailly commented:
Thanks ! I'm waiting to get it to work before merging (b/70993367)

ldemailly committed:
Merge branch 'master' into tsvindex

ldemailly committed:
Merge branch 'tsvindex' of https://github.com/istio/fortio into tsvindex

douglas-reid commented on ui/uihandler.go:
```diff
@@ -324,31 +328,35 @@ func SaveJSON(name string, json []byte) string {
-	url := r.FormValue("url")
-	doRender := (url != "")
+// GetDataList returns the .json files/entries in data dir.
+func GetDataList() (dataList []string) {
```
super nit: just `DataList()`.

douglas-reid commented on ui/uihandler.go:
```diff
@@ -386,10 +394,99 @@ func LogAndAddCacheControl(h http.Handler) http.Handler {
+
+// format for gcloud transfer
+// https://cloud.google.com/storage/transfer/create-url-list
+func sendTsvDataIndex(urlPrefix string, w http.ResponseWriter) {
```
should this be: `sendTSVDataIndex` ?

I think TSV should be all-caps everywhere.

douglas-reid reviewed (commented):
/lgtm

only style nits.

ldemailly committed:
Merge branch 'master' into tsvindex

ldemailly committed:
doug's comments

Though I think gTSVCache is less readable than gTsvCache…
https://laurentsv.com/blog/2017/12/28/about-golang.html

ldemailly committed:
Add CORS header

ldemailly committed:
Return 404s to invalid urls

It was always sending the same full list html before which confuses
scanners

ldemailly committed:
Fixing browse for non report mode

ldemailly commented:
going to merge it because of the `report` mode fixes. even though it doesn't yet quite work on the cloud side and will make follow up PRs as needed as this is reviewed already
